proj/src/main/webapp/chat.js

```diff
@@ -381,13 +380,12 @@ function init() {
 
     const chat = document.getElementById('chatbox');
     chat.addEventListener('scroll', addMoreMessagesAtTheTop);
-
 }
 
 
 function initUserChat(){
-    currentUID = firebase.auth().currentUser.uid;
-    const userTagsRef = firebase.database().ref('/users/'+currentUID+'/allTags');
+    currentUid = firebase.auth().currentUser.uid;
```

Comment:
`const` missing in front. Without it, it might try to assign some global variable of currentUid which may not exist instead of the local one.

Reply:
fixed